collect_app/src/main/java/org/odk/collect/android/formentry/QuitFormDialogFragment.java

```diff
@@ -0,0 +1,134 @@
+package org.odk.collect.android.formentry;
+
+import android.app.Dialog;
+import android.content.Context;
+import android.content.Intent;
+import android.net.Uri;
+import android.os.Bundle;
+import android.widget.ListView;
+
+import androidx.annotation.NonNull;
+import androidx.annotation.Nullable;
+import androidx.annotation.VisibleForTesting;
+import androidx.appcompat.app.AlertDialog;
+import androidx.fragment.app.DialogFragment;
+import androidx.lifecycle.ViewModelProvider;
+
+import com.google.common.collect.ImmutableList;
+
+import org.odk.collect.android.R;
+import org.odk.collect.android.adapters.IconMenuListAdapter;
+import org.odk.collect.android.adapters.model.IconMenuItem;
+import org.odk.collect.android.analytics.Analytics;
+import org.odk.collect.android.application.Collect;
+import org.odk.collect.android.dao.helpers.InstancesDaoHelper;
+import org.odk.collect.android.external.ExternalDataManager;
+import org.odk.collect.android.formentry.audit.AuditEvent;
+import org.odk.collect.android.formentry.saving.FormSaveViewModel;
+import org.odk.collect.android.injection.DaggerUtils;
+import org.odk.collect.android.preferences.AdminKeys;
+import org.odk.collect.android.preferences.AdminSharedPreferences;
+import org.odk.collect.android.utilities.DialogUtils;
+import org.odk.collect.android.utilities.MediaManager;
+
+import java.util.List;
+
+import javax.inject.Inject;
+
+import static android.app.Activity.RESULT_OK;
+
+public class QuitFormDialogFragment extends DialogFragment {
+
+    @Inject
+    Analytics analytics;
+
+    @VisibleForTesting private FormSaveViewModel viewModel;
```

Comment:
Is this needed?

Reply:
Oh sorry I forgot that there. Previously I was thinking of some other approach to use `viewModel` in the tests.